recipes/bdwgc/all/conanfile.py

```diff
@@ -56,88 +56,113 @@ class BdwGcConan(ConanFile):
+                del self.options.fPIC
+            except Exception:
+                pass
+        if Version(self.version) < "8.2.0":
             del self.options.throw_bad_alloc_library
         if not self.options.cplusplus:
-            del self.settings.compiler.libcxx
-            del self.settings.compiler.cppstd
+            try:
+                del self.settings.compiler.libcxx
+            except Exception:
+                pass
+            try:
+                del self.settings.compiler.cppstd
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.settings.os == "Windows":
-            self.requires("libatomic_ops/7.6.10")
+            self.requires("libatomic_ops/7.6.14")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
         for option, _ in self._autotools_options_defaults:
-            self._cmake.definitions["enable_{}".format(option)] = self.options.get_safe(option)
-        self._cmake.definitions["disable_gc_debug"] = not self.options.gc_debug
-        self._cmake.definitions["disable_handle_fork"] = not self.options.handle_fork
-        self._cmake.definitions["install_headers"] = True
-        self._cmake.definitions["build_tests"] = False
-        self._cmake.verbose = True
-        self._cmake.parallel = False
-        self._cmake.configure()
-        return self._cmake
-
-    def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+            if option == "cord":
+                tc.variables["build_cord"] = self.options.get_safe(option)
+            elif option == "cplusplus":
+                tc.cache_variables["enable_cplusplus"] = str(self.options.get_safe(option))
+            else:
+                tc.variables["enable_{}".format(option)] = self.options.get_safe(option)
+        tc.variables["disable_gc_debug"] = not self.options.gc_debug
+        tc.variables["disable_handle_fork"] = not self.options.handle_fork
+        tc.variables["install_headers"] = True
+        tc.variables["build_tests"] = False
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def _extract_copyright(self):
```

Comment:
Since you are the maintainer. maybe add a dedicated license file? much easier to for different ecosystem to rationalize 🙏

Reply:
Yes, agree, upstream master already has it - https://github.com/ivmai/bdwgc/blob/master/LICENSE
It will be released in v8.4.0.